src/nft/components/mint/NFTTotalCost.tsx

```diff
@@ -0,0 +1,125 @@
+import { type ReactNode, useCallback, useMemo, useState } from 'react';
+import { infoSvg } from '../../../internal/svg/infoSvg';
+import { multiplyFloats } from '../../../internal/utils/multiplyFloats';
+import { background, border, cn, text } from '../../../styles/theme';
+import { formatAmount } from '../../../token/utils/formatAmount';
+import { useNFTContext } from '../NFTProvider';
+
+type NFTTotalCostReact = {
+  className?: string;
+  label?: ReactNode;
+};
+
+export function NFTTotalCost({
+  className,
+  label = 'Total cost',
+}: NFTTotalCostReact) {
+  const [isOverlayVisible, setIsOverlayVisible] = useState(false);
+  const { price, mintFee, quantity } = useNFTContext();
+
+  const toggleOverlay = useCallback(() => {
+    setIsOverlayVisible((prev) => !prev);
+  }, []);
+
+  const showOverlay = useCallback(() => {
+    setIsOverlayVisible(true);
+  }, []);
+
+  const hideOverlay = useCallback(() => {
+    setIsOverlayVisible(false);
+  }, []);
+
+  const overlay = useMemo(() => {
+    // only show overlay if mintFee
+    if (
+      price?.amount === undefined ||
+      price?.amountUSD === undefined ||
+      mintFee?.amount === undefined ||
+      mintFee.amountUSD === undefined
+    ) {
+      return null;
+    }
+
+    return (
+      <div
+        className={cn(
+          background.default,
+          border.radius,
+          border.defaultActive,
+          'absolute z-10 w-full border',
+        )}
+      >
+        <div
+          className={cn(
+            'flex items-center justify-between px-4 py-2',
+            text.label2,
+          )}
+        >
+          <div>NFT cost</div>
+          <div>
+            $
+            {formatAmount(`${multiplyFloats(price.amountUSD, quantity)}`, {
+              minimumFractionDigits: 2,
+              maximumFractionDigits: 2,
+            })}
+          </div>
+        </div>
+        <div
+          className={cn(
+            'flex items-center justify-between px-4 py-2',
+            text.label2,
+          )}
+        >
+          <div>Mint fee</div>
+          <div>
+            $
+            {formatAmount(`${multiplyFloats(mintFee?.amountUSD, quantity)}`, {
+              minimumFractionDigits: 2,
+              maximumFractionDigits: 2,
+            })}
+          </div>
+        </div>
+      </div>
+    );
+  }, [mintFee, price, quantity]);
+
+  if (!price?.amount || !price?.currency || !price?.amountUSD) {
+    return null;
+  }
```

Comment:
is there ever a case where an NFT can be free ?

Reply:
yea, the NFTAssetCost component handles that use case, for this total cost component I wasn't showing as we already handle Free elsewhere, but can discuss and revisit.